submissions/Shrai-dev/oop-exercise/index.js

```diff
@@ -0,0 +1,112 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/Shrai-dev/a-tiny-JS-world
+   Web app: https://shrai-dev.github.io/a-tiny-JS-world/
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+class Inhabitant {
+  constructor(species, name, gender, age, saying, friends) {
+    this.species = species;
+    this.name = name;
+    this.gender = gender;
+    this.age = age;
+    this.saying = saying;
+    this.friends = friends;
+  }
+  toString() {
+    return `I am a ${this.species}. My gender is ${this.gender}. My name is ${this.name}. I am ${this.age} years old. My friends are ${this.friends.join(", ")}. I say: ${this.saying}`;
+  }
+}
+
+class Animal extends Inhabitant {
+   constructor(species, name, gender, age, paws, saying, friends) {
+     super(species, name, gender, age, saying, friends);
+     this.paws = paws;
+   }
+   toString() {
+     return super.toString() + ` I have ${this.paws} paws.`;
+   }
+}
+
+class Cat extends Animal {
+  constructor(name, gender, age, friends) {
+    super("cat", name, gender, age, 4, "meow!", friends);
+  }
+  toString() {
+    return super.toString();
+  }
```

Comment:
Every method is inherited just like properties.
If the only purpose of a method overloading is to call an inherited method, then there is no actually point.